metadata/src/main/java/com/google/cloud/teleport/metadata/HelperImage.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (C) 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.metadata;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Annotation that marks a helper image for a Dataflow Template. The helper images will use the same
+ * underlying image as the template they are annotating.
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.TYPE)
+public @interface HelperImage {
```

Comment:
Reading through this PR, I'm -1 on this change at this point in time.

I think that this is probably a premature optimization which is just as likely to cost us time down the line as save it. Specific things I'm worried about:

1) What if we want to release job builder separately from the rest of the templates (which seems reasonable)
2) What if we want job builder to have a different set of dependencies than any other template? I could see this being the case if, for example, we wanted it to contain all ML dependencies (and basically be a fat jar)
3) I don't really see the case for this being generically useful - I think it is unlikely that we'll want to continue to attach other helper images.
4) I already found the container publish flow confusing - this adds an extra layer which makes it harder to understand what we're doing.

I would instead vote for having a separate command (or arg to a top level command) to stage the job server. It could use its own requirements file to install dependencies, it could still use the yaml image (though it doesn't really have to in the future), and it can be run independently.

Basically, I don't think we need to totally couple job builder server release to the yaml template release, and I don't see a ton of advantages to doing so

Reply:
Refactored to use existing `@Template` annotation. Let me know what you think